merkle/merkle_path.go

```diff
@@ -32,13 +32,13 @@ const vvLevel = 4
 // NodeFetch bundles a nodeID with additional information on how to use the node to construct the
 // correct proof.
 type NodeFetch struct {
-	NodeID storage.NodeID
+	ID     compact.NodeID
```

Comment:
We're losing some error checking here because the storage code has checks on creating IDs e.g that they're inside the tree depth whereas the compact code accepts any values. I suppose almost the exact same checks are done when the fetches are converted so perhaps this isn't important.

Reply:
Well yes but you can create compact node IDs with level > 64. It's often better to prevent the creation of invalid objects rather than detect them every time they're used. This is easier in languages that handle errors better than Go does.

The second part is what makes this probably OK as I don't think it creates any new opportunities for things to go wrong.